src/Latte/Macros/BlockMacros.php

```diff
@@ -294,18 +295,14 @@ public function macroBlockEnd(MacroNode $node, PhpWriter $writer)
 				$node->content = $node->innerContent;
 			}
 
+			if (($node->name === 'snippet' || $node->name === 'snippetArea') && strpos($node->data->name, '$') === FALSE) {
+				$type = $node->name === 'snippet' ? 'TYPE_STATIC' : 'TYPE_AREA';
+				$node->content = '<?php $this->global->snippetDriver->enter('
```

Comment:
I know, but it is better to have readable source code that generated source code.